src/python/WMCore/Database/DBTest.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+#pylint: disable=
+"""
+File       : DBTest.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: Base class for WMCore database related unit tests.
+             It defines test database name based on user preferences specified via
+             environment variables and setup logic to setup/tear down actions
+             based on database.
+"""
+from __future__ import print_function
+
+# system modules
+import os
+import sys
+import unittest
+
+class DBTest(unittest.TestCase):
+    def __init__(self, methodName='runTest'):
+        super(WMCoreUnitTest, self).__init__(methodName)
+        self.dbName = os.environ.get('WMCORE_TEST_DATABASE', \
+                'unittest_%s' % self.__class__.__name__)
+        print("WMCoreTestDB test %s database" % self.dbName)
+        enforce = os.environ.get('WMCORE_TEST_DATABASE_DELETE', False)
+        if  enforce:
+            print("WMCORE_TEST_DATABASE_DELETE=%s, will delete %s" % (enforce, self.dbName))
+            self.deleteDatabase()
+            self.createDatabase()
+
+    def setUp(self):
+        """Setup database for unittests"""
+        if  self.dbName.startswith('unittest'):
```

Comment:
Well, probably this will serve better:

```
if self.dbName == 'unittest_%s' % self.__class__.__name__:
```

if you worry about collision from environment.


On  0, Eric Vaandering <notifications@github.com> wrote:
> > +
> > +class DBTest(unittest.TestCase):
> > +    def __init__(self, methodName='runTest'):
> > +        super(WMCoreUnitTest, self).__init__(methodName)
> > +        self.dbName = os.environ.get('WMCORE_TEST_DATABASE', \
> > +                'unittest_%s' % self.__class__.__name__)
> > +        print("WMCoreTestDB test %s database" % self.dbName)
> > +        enforce = os.environ.get('WMCORE_TEST_DATABASE_DELETE', False)
> > +        if  enforce:
> > +            print("WMCORE_TEST_DATABASE_DELETE=%s, will delete %s" % (enforce, self.dbName))
> > +            self.deleteDatabase()
> > +            self.createDatabase()
> > +
> > +    def setUp(self):
> > +        """Setup database for unittests"""
> > +        if  self.dbName.startswith('unittest'):
> 
> Why did you choose here to not start with unittest_ or even just check for `'unittest_%s' % self.__class__.__name__`?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/6283/files#r40341251